ckanext/datastore/blueprint.py

```diff
@@ -292,6 +305,19 @@ def dump_to(
     elif fmt == 'xml':
         writer_factory = xml_writer
         records_format = 'objects'
+    # (canada fork only): psql dump format
+    elif fmt == 'sql':
+        datastore_uri = str(DatastoreBackend.get_active_backend()._get_write_engine().url)
+        # dump clean table, schema, and data.
+        # quote all for blank values.
+        cmd = subprocess.Popen(
+            ['pg_dump', datastore_uri, '--clean', '--if-exists', '--disable-triggers',
+             '--no-owner', '--quote-all-identifiers', '-t', resource_id],
+            stdout=subprocess.PIPE)
+        def stream_sql(process):
+            for c in iter(lambda: process.stdout.read(-1), b""):
```

Comment:
We will want some maximum buffer size to prevent getting the web process killed for using too much RAM.

Reply:
will just use `io.DEFAULT_BUFFER_SIZE` to be super safe.